validator/distro_validator_test.go

```diff
@@ -266,3 +266,59 @@ func TestEnsureNonZeroID(t *testing.T) {
 	assert.Nil(ensureHasNonZeroID(ctx, &distro.Distro{Id: "foo"}, conf))
 	assert.Nil(ensureHasNonZeroID(ctx, &distro.Distro{Id: " "}, conf))
 }
+
+func TestEnsureValidContainerPool(t *testing.T) {
+	assert := assert.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	db.SetGlobalSessionProvider(testutil.TestConfig().SessionFactory())
+	assert.NoError(db.Clear(distro.Collection))
+
+	conf := &evergreen.Settings{
+		ContainerPools: evergreen.ContainerPoolsConfig{
+			Pools: []evergreen.ContainerPool{
+				evergreen.ContainerPool{
+					Distro: "d4",
+					Id:     "test-pool-valid",
+				},
+				evergreen.ContainerPool{
+					Distro: "d1",
+					Id:     "test-pool-invalid",
+				},
+			},
+		},
+	}
+
+	d1 := &distro.Distro{
```

Comment:
just a heads up that whenever my PR gets merged in, this will fail because a valid distro will also need to have `NoTasksAllowed` set to `true`. i'll let you know when that comes.